source/configuration/modules/omfile.rst

```diff
@@ -444,86 +451,57 @@ Caveats/Known Bugs
 
-::
+Currently, there are two providers available ":doc:`gt <sigprov_gt>`" and ":doc:`ksi <sigprov_ksi>`". 
 
-  action(type="omfile" dirCreateMode="0700" FileCreateMode="0644"
-         File="/var/log/messages")
+**TODO `Sign log messages through signature provider Guardtime <http://www.rsyslog.com/how-to-sign-log-messages-through-signature-provider-guardtime/>`_**
 
-Legacy Configuration Directives
--------------------------------
+=============  ==============
+cry.provider:  "providername"
+Default:       none
+=============  ==============
 
-Note that the legacy configuration parameters do **not** affect
-new-style action definitions via the action() object. This is
-by design. To set default for action() objects, use module parameters
-in the
+Selects a cryptographic provider for **log encryption**. By selecting a provider, the encryption feature is turned on.
 
-::
+Currently, there only is one provider called ":doc:`gcry <../cryprov_gcry>`".
 
-  module(load="builtin:omfile" ...)
-
-object.
-
-Read about :ref:`the importance of order in legacy configuration<legacy-action-order>`
-to understand how to use these configuration directives.
-**Legacy directives should NOT be used when writing new configuration files.**
-
--  **$DynaFileCacheSize**
-   equivalent to the "dynaFileCacheSize" parameter
--  **$OMFileZipLevel**
-   equivalent to the "zipLevel" parameter
--  **$OMFileFlushInterval**
-   equivalent to the "flushInterval" parameter
--  **$OMFileASyncWriting**
-   equivalent to the "asyncWriting" parameter
--  **$OMFileFlushOnTXEnd**
-   equivalent to the "flushOnTXEnd" parameter
--  **$OMFileIOBufferSize**
-   equivalent to the "IOBufferSize" parameter
--  **$DirOwner**
-   equivalent to the "dirOwner" parameter
--  **$DirGroup**
-   equivalent to the "dirGroup" parameter
--  **$FileOwner**
-   equivalent to the "fileOwner" parameter
--  **$FileGroup**
-   equivalent to the "fileGroup" parameter
--  **$DirCreateMode**
-   equivalent to the "dirCreateMode" parameter
--  **$FileCreateMode**
-   equivalent to the "fileCreateMode" parameter
--  **$FailOnCHOwnFailure**
-   equivalent to the "failOnChOwnFailure" parameter
--  **$F$OMFileForceCHOwn**
-   equivalent to the "ForceChOwn" parameter
--  **$CreateDirs**
-   equivalent to the "createDirs" parameter
--  **$ActionFileEnableSync**
-   equivalent to the "enableSync" parameter
--  **$ActionFileDefaultTemplate**
-   equivalent to the "template" module parameter
--  **$ResetConfigVariables**
-   Resets all configuration variables to their default value.
+Examples
+********
+
+::
 
-Legacy Sample
-^^^^^^^^^^^^^
+  # Write to myfile
+  action(type="omfile" dirCreateMode="0700" fileCreateMode="0644" file="/var/log/myfile")
 
-The following command writes all syslog messages into a file.
+Legacy Configuration
+====================
 
 ::
 
+  # Write all mensajes into /var/log/messages
   $DirCreateMode 0700
   $FileCreateMode 0644
   *.* /var/log/messages
+  
```

Comment:
shhould have an example with template format.

This is one of the modules where the legacy format gets used a lot (and if you are not needing to set anything other than file/dynafile/template is arguably clearer in the legacy format). So it may be worth putting the simple legacy example very early in the file.